tests/unit/test_store/fake_storage.py

```diff
@@ -1,107 +1,244 @@
-        else:
-            return None
-
-    def put_features(self, features_name: str, content: pd.DataFrame):
-        self._store[features_name] = content
-
-
-class FakeTripleStore(TripleStoreABC):
-    def with_query(self, sparql_query: str, substitution_variables: dict = None,
-                   sparql_prefixes: str = "") -> 'TripleStoreABC':
-        return self
-
-    def with_query_from_file(self, sparql_query_file_path: str, substitution_variables: dict = None,
-                             prefixes: str = "") -> 'TripleStoreABC':
-        return self
-
-    def get_dataframe(self) -> pd.DataFrame:
-        return pd.DataFrame([{"col1": "A", "col2": "B"}])
+import copy
+import io
+from http.client import HTTPResponse
+from typing import Any, Optional, Union, Collection, MutableMapping
+
+from elasticsearch import Elasticsearch
+from es_pandas import es_pandas
+
+from sem_covid.adapters.abstract_store import *
+from minio import Minio
+
+
+class FakeMinioObject:
+
+    def __init__(self, object_name: str, content):
+        self.object_name = object_name
+        self.content = content
+
+
+class FakeHTTPMinioResponse:
+    def __init__(self, content: io.BytesIO = None):
+        self.content = content
+
+    def read(self):
+        if self.content is not None:
+            return self.content.read()
+        return self.content
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        pass
+
+
+class FakeMinioClient(Minio):
+
+    def __init__(self):
+        self.storage = dict()
+
+    def put_object(self, bucket_name, object_name, data, length, content_type="application/octet-stream", metadata=None,
+                   sse=None, progress=None, part_size=0, num_parallel_uploads=3, tags=None, retention=None,
+                   legal_hold=False):
+        self.storage[bucket_name][object_name] = FakeMinioObject(object_name, data)
+
+    def remove_objects(self, bucket_name, delete_object_list, bypass_governance_mode=False):
+        for tmp_object in delete_object_list:
+            self.storage[bucket_name].pop(tmp_object._name, None)
+        return list()
+
+    def make_bucket(self, bucket_name, location=None, object_lock=False):
+        self.storage[bucket_name] = dict()
+
+    def get_object(self, bucket_name, object_name, offset=0, length=0, request_headers=None, ssec=None, version_id=None,
+                   extra_query_params=None):
+        if bucket_name in self.storage.keys():
+            if object_name in self.storage[bucket_name].keys():
+                return FakeHTTPMinioResponse(self.storage[bucket_name][object_name].content)
+        return FakeHTTPMinioResponse()
+
+    def bucket_exists(self, bucket_name):
+        return bucket_name in self.storage.keys()
+
+    def list_objects(self, bucket_name, prefix=None, recursive=False, start_after=None, include_user_meta=False,
+                     include_version=False, use_api_v1=False):

```

Comment:
this is quite an implementation on its own! 
probably we do not need so advance fakes :) Next time we can try together to find a simplified fake implementation :D